api/src/main/java/ru/urururu/sanity/api/InstructionParser.java

```diff
@@ -0,0 +1,158 @@
+        } else {
+            returnType = target.getType().getElementType().getReturnType();
+        }
+
+        LValue lvalue = returnType.isVoid() ? null : ctx.getOrCreateTmpVar(instruction);
+
+        List<RValue> args = new ArrayList<>();
+        for (V param : parameters) {
+            args.add(parsers.parseRValue(ctx, param));
+        }
+
+        return new Call(
+                target,
+                lvalue,
+                args,
+                parsers.getSourceRange(instruction)
+        );
+    }
+
+    protected Cfe createBinaryAssignment(Ctx ctx, I instruction, V left, BinaryExpression.Operator operator, V right) {
+        return new Assignment(
+                ctx.getOrCreateTmpVar(instruction),
+                new BinaryExpression(
+                        parsers.parseRValue(ctx, left),
+                        operator,
+                        parsers.parseRValue(ctx, right)
+                ),
+                parsers.getSourceRange(instruction)
+        );
+    }
+
+    protected RValue getPointer(Ctx ctx, V basePointer, Iterable<V> indices) {
+        RValue pointer = parsers.parseRValue(ctx, basePointer);
+
+        for (V index : indices) {
+            pointer = getPointer(pointer, parsers.parseRValue(ctx, index));
+        }
+
+        return pointer;
+    }
+
+    protected RValue getPointer(RValue basePointer, RValue index) {
+        if (basePointer.getType().getElementType() != null) {
+            return new GetElementPointer(basePointer, index);
+        }
+        if (index instanceof Const) {
+            int intIndex = (int) ((Const) index).getValue();
+            Type fieldType = basePointer.getType().getFieldType(intIndex);
+            if (fieldType != null) {
+                return new GetFieldPointer(basePointer, intIndex);
+            }
+        }
+        throw new IllegalStateException("can't index " + CfePrinter.printValue(basePointer) + " by " + index);
+    }
+
+    protected Cfe createSwitch(Ctx ctx, I instruction, V controlValue, V defaultCase, Iterable<V> values, Iterable<V> labels) {
+        Map<RValue, Cfe> cases = FinalMap.createLinkedHashMap();
+
+        Iterator<V> valuesIterator = values.iterator();
+        Iterator<V> labelsIterator = labels.iterator();
+        while (valuesIterator.hasNext() && labelsIterator.hasNext()) {
+            cases.put(parsers.parseRValue(ctx, valuesIterator.next()), ctx.getLabel(labelsIterator.next()));
+        }
+
+        if (valuesIterator.hasNext()) {
+            throw new IllegalStateException("out of labels, but have value");
+        }
+        if (labelsIterator.hasNext()) {
+            throw new IllegalStateException("out of values, but have label");
+        }
+
+        return new Switch(parsers.parseRValue(ctx, controlValue), ctx.getLabel(defaultCase), cases, parsers.getSourceRange(instruction));
+    }
+
+    protected Cfe createGoto(Ctx ctx, I instruction, V label) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Remove this unused method parameter "instruction". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1172)